test/webdav_test.dart

```diff
@@ -0,0 +1,235 @@
+import 'dart:convert';
+import 'dart:io';
+
+import 'package:nextcloud/nextcloud.dart';
+import 'package:test/test.dart';
+
+import 'config.dart';
+
+void main() {
+  final client = getClient();
+
+  group('WebDav', () {
```

Comment:
```suggestion
  group('WebDav', () {
    setUpAll(() async {
      try {
        await client.webDav.delete(Config.testDir);
        // ignore: empty_catches
      } on RequestException catch (ex) {
        if (ex.statusCode != 404) {
          rethrow;
        }
      }
    });
    
```
To fix repeatability of tests. Otherwise all good. :+1:

Reply:
In theory we could remove `testDir` from the `Config` since it only concerns the webDav tests.

Reply:
I also though about removing it from the tests, because everything can be done in the main directory, because it's only a development instance that resets automatically.

Reply:
I think I'll do that with https://github.com/Viktoriaschule/dart-nextcloud/issues/24

Reply:
Btw I removed the code you suggested, because the test instance is always clean when starting the tests.

Reply:
@vauvenal5 do you agree so I can merge?

Reply:
Yes, the test instance is but the development instance is not after the first run of the tests.

Reply:
Hm if someone is using the development instance the are probably also using the test instance, but to be sure I can add it, yes.